tests/test_fields.py

```diff
@@ -0,0 +1,31 @@
+import sys
+from dataclasses import InitVar
+
+import pytest
+
+import pydantic.dataclasses
+from pydantic import fields
+
+
+def test_field_info_annotation_keyword_argument():
+    """This tests that `FieldInfo.from_field` raises an error if passed the `annotation` kwarg.
+
+    At the time of writing this test there is no way `FieldInfo.from_field` could receive the `annotation` kwarg from
+    anywhere inside Pydantic code. However, it is possible that this API is still being in use by applications and
+    third-party tools.
+    """
+    with pytest.raises(TypeError) as e:
+        fields.FieldInfo.from_field(annotation=())
+
+    assert e.value.args == ('"annotation" is not permitted as a Field keyword argument',)
```

Comment:
If we're going to just raise an error, I'm not sure there's much value in providing a custom error if we can't offer a solution / link to workaround.